File: backend/src/main/java/com/godeltech/mastery/backend/service/impl/MaintenanceServiceImpl.java
package com.godeltech.mastery.backend.service.impl;

import com.godeltech.mastery.backend.domain.dto.responce.MaintenanceDTO;
import com.godeltech.mastery.backend.mapper.MaintenanceMapper;
import com.godeltech.mastery.backend.repository.MaintenanceRepository;
import com.godeltech.mastery.backend.service.MaintenanceService;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class MaintenanceServiceImpl implements MaintenanceService {

  private final MaintenanceRepository maintenanceRepository;
  private final MaintenanceMapper maintenanceMapper;

  @Override
  public List<MaintenanceDTO> getAllMaintenance() {
    return maintenanceMapper.toDTOList(maintenanceRepository.findAll());
  }
}

File: backend/src/main/kotlin/com/godeltech/mastery/backend/service/impl/MaintenanceServiceImpl.kt
package com.godeltech.mastery.backend.service.impl

import com.godeltech.mastery.backend.domain.dto.responce.MaintenanceDto
import com.godeltech.mastery.backend.domain.entity.Maintenance
import com.godeltech.mastery.backend.repository.MaintenanceRepository
import com.godeltech.mastery.backend.service.MaintenanceService
import org.springframework.stereotype.Service

@Service
class MaintenanceServiceImpl(private val maintenanceRepository: MaintenanceRepository) : MaintenanceService {

    override fun getAllMaintenance(): List<MaintenanceDto> = maintenanceRepository.findAll().map(Maintenance::toDto)
}